amp rotate PR, align score and exponential joysticks

zabzzz opened the pull request.

zabzzz committed:
Align score based off robot angle from AprilTag

zabzzz committed:
Spaghetti code clean up

zabzzz committed:
PivotPIDCommand rename

felt like a mouthful, and too ugly looking so renamed to "PivotPID"

zabzzz committed:
.

.

minhd-vu commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -31,59 +31,55 @@ public final class Constants {
     public static final double PIVOT_THRESHOLD = 0.03;
+    
+    // Others
     public static final double pivotMaxVelocity = 4;
     public static final double pivotMaxAccel = 10;
```
why are these not in CAPS

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -46,34 +47,27 @@ public class RobotContainer {
-                getDriverLeftY(),
-                getDriverLeftX(),
-                getDriverRightX(),
+                (.5)*Math.pow(getDriverLeftY(), 5) + (.5)*getDriverLeftY(),
```
make these values a constant so that it can easily be adjusted if needed

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -106,25 +100,26 @@ public static double getDriverRightX() {
     m_driverController.rightTrigger().onTrue(new AmpTrapScore(ManipulatorConstants.PIVOT_AMP_GOAL));
-    m_driverController.button(OIConstants.RIGHT_BUMPER).onTrue(new ScoreGoal(ManipulatorConstants.PIVOT_FAR_SCORE));
-    m_driverController.button(OIConstants.LEFT_BUMPER).onTrue(new ScoreGoal(ManipulatorConstants.PIVOT_CLOSE_SCORE));
+    m_driverController.rightBumper().onTrue(new ScoreGoal(ManipulatorConstants.PIVOT_FAR_SCORE));
```
dope

minhd-vu commented on src/main/java/frc/robot/commands/vision/AlignScore.java:
```diff
@@ -1,54 +1,55 @@
+  public void execute() {
+    // angle = (RobotContainer.limeSubsystem.getHorizontalAngle()-(Math.PI/2)) + RobotContainer.swerveDrive.getHeading().getRadians();
+    angle = RobotContainer.limeSubsystem.getHorizontalAngle();
+    if (angle < -VisionConstants.threshold || angle > VisionConstants.threshold){
```
```suggestion
    if (Math.abs(angle) > VisionConstants.threshold){
```

minhd-vu commented on src/main/java/frc/robot/commands/vision/AlignScore.java:
```diff
@@ -1,54 +1,55 @@
+            true, true);
+    } else {
+      RobotContainer.swerveDrive.drive(
+        (.5)*Math.pow(RobotContainer.getDriverLeftY(), 5) + (.5)*RobotContainer.getDriverLeftY(),
```
why not put these curves inside the `getDriverLeft...()` methods? Lots of duplication here.

minhd-vu commented on src/main/java/frc/robot/commands/vision/AlignScore.java:
```diff
@@ -1,54 +1,55 @@
+            true, true);
+    }
+  }
+    // insert code to adjust robot angle here
```
delete comment

minhd-vu commented on src/main/java/frc/robot/subsystems/LimelightSubsystem.java:
```diff
@@ -30,7 +31,10 @@ public void turnOnLEDs() {
-    return x;
+    // roboPose = RobotContainer.swerveDrive.getPose();
+    // angle = 1/Math.tan(roboPose.getX()/roboPose.getY());
+    angle = x;
+    return angle;
```
```suggestion
    return x;
```

minhd-vu reviewed (changes requested):
good work just a little nits

zabzzz committed:
PR addressed fixes

zabzzz committed:
getInputwithDeadband

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,40 +85,61 @@ public RobotContainer() {
    * {@link JoystickButton}.
    */
 
+  public static double getLeftXwithDeadband() {
```
```suggestion
  public static double getLeftXWithDeadband() {
```

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,40 +85,61 @@ public RobotContainer() {
+      (.5)*(getLeftXwithDeadband());
+  }
+
+  public static double getLeftYwithDeadband() {
```
```suggestion
  public static double getLeftYWithDeadband() {
```

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,40 +85,61 @@ public RobotContainer() {
 
-  public static double getDriverRightX() {
-    return -MathUtil.applyDeadband(m_driverController.getRightX(), OIConstants.kDriveDeadband);
+  public static double getRightXwithDeadband() {
```
```suggestion
  public static double getRightXWithDeadband() {
```

zabzzz committed:
limelight - drivetrain abstraction

- Limelight blink on game piece acquisition
- Drivetrain commanding abstracted to separate methods

minhd-vu commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -91,40 +85,61 @@ public RobotContainer() {
+
   public static double getDriverLeftX() {
-    return -MathUtil.applyDeadband(m_driverController.getLeftX(), OIConstants.kDriveDeadband);
+    return (.5)*Math.pow(
+      (getLeftXWithDeadband()), 5) + 
+      (.5)*(getLeftXWithDeadband());
```
```suggestion
    return 0.5 * Math.pow(getLeftXWithDeadband(), 5) + 0.5 * getLeftXWithDeadband();
```
Did vscode format it like this? Try the format in the suggestion, looks cleaner imo.

zabzzz committed:
fix

zabzzz committed:
fix.2

zabzzz committed:
Merge remote-tracking branch 'origin/main' into zabz-main
